src/Microsoft.ML.Core/Utilities/Utils.cs

```diff
@@ -634,7 +634,7 @@ public static bool IsSorted(Float[] values)
 
             for (int i = 1; i < values.Length; i++)
             {
-                if (!(values[i] >= prev))
+                if (values[i] < prev)
                     return false;
```

Comment:
This is not an equivalent change.
Note that `float.NaN` is a special value that is allowed throughout ML.NET (a NaN represents a missing value)

The NaN has the property that it is not LESS, not EQUAL and not MORE than any other value: all arithmetic comparisons that include a NaN are `false`. 
So, we made a decision that, in the context of `IsSorted`, a presense of at least one NaN makes the array NOT sorted.

With your change though, the presense of NaNs not allow for `IsSorted` to satill be true, it can actually make a non-sorted array seemingly sorted:

In the old code, `IsSorted(new[] {3f, float.NaN, 1f}) == false`
With your changes, `IsSorted(new[] {3f, float.NaN, 1f}) == true`

We can argue about how to handle missing values for the purpose of sorting, but obviously the `[3, NaN, 1]` array should not be considered 'sorted' under any definition of 'sorted' :)

Reply:
Incidentally if you search our codebase for the regular expression `!\([^\)]+>` or `!\([^\)]+<`, you will find perhaps some hundred or so examples of this pattern which, as @Zruty0 points out, is deliberate and absolutely essential for the correct handling of floating point numbers.

Reply:
Maybe it would be a good idea to:

add a comment here describing why the code is like it is.
If that is duplicated in a few places, extract a common method that these places call, and put the comment on the common method.